### PR TITLE
Copy CC methods to CM

### PR DESCRIPTION
**Pull request by quantumagi**
See issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/1797.

### CODE REVIEW

**Comment by quantumagi** [comment]
Nice - only some PoA integration tests are failing,

image

**Comment by MithrilMan** [comment]
can you move the class to its own file?

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 1]
```diff
@@ -21,6 +22,168 @@
 
 namespace Stratis.Bitcoin.Consensus
 {
+    internal class BlockIndexer
+    {
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: I think both dictionaries are not needed, for `blocksById ` we can reuse the dictionary in CHT
And for `blocksByHeight ` we can just leverage the count of that same dictionary (genesis is at position zero anyway so there is not even an offset by 1)

Comment (MithrilMan) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: yes basically `private readonly Dictionary<int, ChainedHeader> blocksByHeight` could be a
`private readonly List<ChainedHeader> blocksByHeight` where the index is the height
we have just to be sure to populate it in order

File: src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs [hunk 2]
```diff
@@ -286,6 +296,23 @@ public ChainedHeaderBlock GetChainedHeaderBlock(uint256 blockHash)
             return chainedHeaderBlock;
         }
 
+        /// <inheritdoc />
+        public ChainedHeaderBlock GetChainedHeaderBlock(int blockHeight)
+        {
+            ChainedHeaderBlock chainedHeaderBlock = null;
+            if (this.chainedHeadersByHash.Count <= blockHeight)
+            {
+                KeyValuePair<uint256, ChainedHeader> chainedHeaderItem = this.chainedHeadersByHash.ElementAt(blockHeight);
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs]: `ElememtAt` is not a good idea to use int his case, I will reissue a second PR.

**Comment by dangershony** [comment]
Closing this PR and will create another one in its place.

**Pull request closed without merging**